library/blivet.py

```diff
@@ -665,6 +669,35 @@ def _get_size(self):
             size = self._blivet_pool._device.align(size, roundup=True)
         return size
 
+    def _detach_cache(self):
+        """ Detach cache from the volume and remove the unused cache pool """
+        try:
+            cpool_name = self._device.cache.detach()
+        except Exception as e:
+            raise BlivetAnsibleError("failed to detach cache from volume '%s': %s" % (self._device.name, str(e)))
+
+        # full reset is needed for the cache pool to be added to the devicetree so we can remove it
+        self._blivet.reset()
+
+        cpool_name = cpool_name.rstrip("_cpool")
+        cpool_device = self._blivet.devicetree.resolve_device("%s-%s" % (self._device.vg.name, cpool_name))
+
+        self._blivet.destroy_device(cpool_device)
+
+    def _attach_cache(self):
+        """ Create a new cache pool and attach it to the volume """
+        raise BlivetAnsibleError("adding cache to an existing volume is currently not supported")
```

Comment:
Blivet unfortunately doesn't support creating a new cache pool. We can attach an existing cache pool to the LV, but we can't create it. I plan to add support for creating cache pools to blivet and I'll update this when the support is available.